tests/MyGame/Example/Monster.go

```diff
@@ -6,9 +6,461 @@ import (
 	flatbuffers "github.com/google/flatbuffers/go"
 
 	MyGame "MyGame"
+	MyGame__Example2 "MyGame/Example2"
 )
 
 /// an example documentation comment: monster object
+type MonsterT struct {
+	Pos *Vec3T
+	Mana int16
+	Hp int16
+	Name string
+	Inventory []byte
+	Color Color
+	Test *AnyT
+	Test4 []*TestT
+	Testarrayofstring []string
+	Testarrayoftables []*MonsterT
+	Enemy *MonsterT
+	Testnestedflatbuffer []byte
+	Testempty *StatT
+	Testbool bool
+	Testhashs32Fnv1 int32
+	Testhashu32Fnv1 uint32
+	Testhashs64Fnv1 int64
+	Testhashu64Fnv1 uint64
+	Testhashs32Fnv1a int32
+	Testhashu32Fnv1a uint32
+	Testhashs64Fnv1a int64
+	Testhashu64Fnv1a uint64
+	Testarrayofbools []bool
+	Testf float32
+	Testf2 float32
+	Testf3 float32
+	Testarrayofstring2 []string
+	Testarrayofsortedstruct []*AbilityT
+	Flex []byte
+	Test5 []*TestT
+	VectorOfLongs []int64
+	VectorOfDoubles []float64
+	ParentNamespaceTest *MyGame.InParentNamespaceT
+	VectorOfReferrables []*ReferrableT
+	SingleWeakReference uint64
+	VectorOfWeakReferences []uint64
+	VectorOfStrongReferrables []*ReferrableT
+	CoOwningReference uint64
+	VectorOfCoOwningReferences []uint64
+	NonOwningReference uint64
+	VectorOfNonOwningReferences []uint64
+	AnyUnique *AnyUniqueAliasesT
+	AnyAmbiguous *AnyAmbiguousAliasesT
+	VectorOfEnums []Color
+}
+
+func MonsterPack(builder *flatbuffers.Builder, t *MonsterT) flatbuffers.UOffsetT {
+	if t == nil { return 0 }
+	nameOffset := builder.CreateString(t.Name)
+	inventoryOffset := flatbuffers.UOffsetT(0)
+	if t.Inventory != nil {
+		inventoryOffset = builder.CreateByteString(t.Inventory)
+	}
+	testOffset := flatbuffers.UOffsetT(0)
+	if t.Test != nil {
+		switch t.Test.Type {
+		case AnyMonster:
+			testOffset = MonsterPack(builder, t.Test.Value.(*MonsterT))
+		case AnyTestSimpleTableWithEnum:
+			testOffset = TestSimpleTableWithEnumPack(builder, t.Test.Value.(*TestSimpleTableWithEnumT))
+		case AnyMyGame_Example2_Monster:
+			testOffset = MyGame__Example2.MonsterPack(builder, t.Test.Value.(*MyGame__Example2.MonsterT))
+		}
+	}
+	test4Offset := flatbuffers.UOffsetT(0)
+	if t.Test4 != nil {
+		test4Length := len(t.Test4)
+		MonsterStartTest4Vector(builder, test4Length)
+		for j := test4Length - 1; j >= 0; j-- {
+			TestPack(builder, t.Test4[j])
+		}
+		test4Offset = builder.EndVector(test4Length)
+	}
+	testarrayofstringOffset := flatbuffers.UOffsetT(0)
+	if t.Testarrayofstring != nil {
```

Comment:
This generated code is generally quite large.. I wonder if any of this can be factored out into helper methods. But I guess without templates that is hard..

Reply:
We can create helper methods for the default types in the flatbuffers package but without templates we are just moving code from the generated file to the std library.

Everything else than default types we have to generate code.